src/Fmt_ast.ml

```diff
@@ -132,18 +132,27 @@ let rec sugar_arrow_typ c ({ast= typ} as xtyp) =
   | Ptyp_arrow (l, t1, t2) ->
       Cmts.relocate c.cmts ~src:ptyp_loc ~before:t1.ptyp_loc
         ~after:t2.ptyp_loc ;
-      (l, sub_typ ~ctx t1) :: sugar_arrow_typ c (sub_typ ~ctx t2)
+      let rest =
+        match t2.ptyp_attributes with
+        | [] -> sugar_arrow_typ c (sub_typ ~ctx t2)
+        | _ -> [(Nolabel, sub_typ ~ctx t2)]
+      in
+      (l, sub_typ ~ctx t1) :: rest
   | _ -> [(Nolabel, xtyp)]
 
 let rec sugar_class_arrow_typ c ({ast= typ} as xtyp) =
   let ctx = Cty typ in
-  let {pcty_desc; pcty_loc} = typ in
+  let {pcty_desc; pcty_loc; pcty_attributes} = typ in
   match pcty_desc with
   | Pcty_arrow (l, t1, t2) ->
       Cmts.relocate c.cmts ~src:pcty_loc ~before:t1.ptyp_loc
         ~after:t2.pcty_loc ;
-      (l, `core_type (sub_typ ~ctx t1))
-      :: sugar_class_arrow_typ c (sub_cty ~ctx t2)
+      let rest =
+        match pcty_attributes with
+        | [] -> sugar_class_arrow_typ c (sub_cty ~ctx t2)
```

Comment:
In the other cases, it is the attributes of the argument passed to `sugar_` that are checked for emptiness. This case is different, intentionally?

Reply:
it is not, thanks